### PR TITLE
Introduced 'discoveryurl' config property. Added discovering Hue brid…

### PR DESCRIPTION
**Pull request by apereverzin**
Introduced 'discoveryurl' config property. 

Added discovering Hue bridge via 'https://www.meethue.com/api/nupnp' link. 

Hue bridge discovery logic moved to HueBridgeDiscoverer class.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridgeDiscoverer.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
+package org.openhab.binding.hue.internal;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridgeDiscoverer.java]: add license header

File: bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridgeDiscoverer.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import org.openhab.binding.hue.internal.data.BridgeData;
+import org.openhab.binding.hue.internal.data.HueSettings;
+import org.openhab.binding.hue.internal.hardware.HueBridge;
+import org.openhab.binding.hue.internal.tools.JSONTransformer;
+import org.openhab.binding.hue.internal.tools.SsdpDiscovery;
+import org.osgi.service.cm.ConfigurationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridgeDiscoverer.java]: please add a short description what this class is about

File: bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/data/BridgeData.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package org.openhab.binding.hue.internal.data;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/data/BridgeData.java]: add license header

File: bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/data/BridgeData.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.json.simple.JSONObject;
+
+/**
+ * Encapsulates HUE bridge attributes.
+ * 
+ * @author andreypereverzin
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/data/BridgeData.java]: add @version tag

File: bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/tools/JSONTransformer.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
+package org.openhab.binding.hue.internal.tools;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/tools/JSONTransformer.java]: add license header

File: bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/tools/JSONTransformer.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+
+import org.json.simple.JSONArray;
+import org.json.simple.parser.ContentHandler;
+import org.json.simple.parser.ParseException;
+
+/**
+ *
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/tools/JSONTransformer.java]: add @version tag and a short explanation what this class is about

**Comment by teichsta** [comment]
Thanks for this contribution @apereverzin and my apologies for not coming back to earlier to you. I've some review comments inline. Please note that 1.8 [will be the last release with a 1.x core](https://community.openhab.org/t/merry-christmas-and-a-brief-outlook-on-2016/5333) and that there is a completely rewritten Hue-Binding available for Eclipse Smarthome.

Could you please finally squash the commits into after incorporating my feedback.

Thanks again, Thomas E.-E.

**Comment by teichsta** [comment]
any chance to incorporate the changes until tonight? Would move this PR to 1.9 otherwise.

**Comment by teichsta** [comment]
moved to 1.9

**Comment by apereverzin** [comment]
Hi Thomas
Unfortunately, I cannot do this in the nearest days.

**Comment by teichsta** [comment]
> Unfortunately, I cannot do this in the nearest days.

no worries … just send your changes when they are ready.

**Comment by 9037568** [comment]
Any updates on this, @apereverzin?

**Pull request closed without merging**